compiler/rustc_infer/src/infer/canonical/query_response.rs

```diff
@@ -360,25 +409,35 @@ impl<'tcx> InferCtxt<'tcx> {
             original_values, query_response,
         );
 
-        let mut value = self.query_response_instantiation_guess(
-            cause,
-            param_env,
-            original_values,
-            query_response,
-        )?;
+        let (result_args, opaques) =
+            self.query_response_instantiation_guess(cause, original_values, query_response);
+
+        let mut obligations = vec![];
+
+        for (key, hidden) in opaques {
+            obligations.extend(
+                self.at(cause, param_env)
+                    .eq(
+                        DefineOpaqueTypes::Yes,
+                        Ty::new_opaque(self.tcx, key.def_id.into(), key.args),
+                        hidden,
+                    )?
+                    .obligations,
+            )
+        }
```

Comment:
all non-nll query callers still need their opaque types registered. After #122077 this will become a simple `register_hidden_type` call